### ISSUE TITLE
`DistanceBand` failing when `ids` passed

### ISSUE THREAD

**Issue by darribas** [pysal/pysal#945]
The following example:

```python
In [43]: import pysal as ps

In [44]: from pysal.contrib import pdio

In [45]: db = pdio.read_files(ps.examples.get_path('columbus.shp'))

In [46]: xys = np.array(db.geometry.apply(lambda i: i.centroid).tolist())

In [47]: w = ps.weights.DistanceBand(xys, 100, ids=db.POLYID)
```

Produces the following error:

```python
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-47-6bb7b334aa26> in <module>()
----> 1 w = ps.weights.DistanceBand(xys, 100, ids=db.POLYID)

/home/dani/anaconda/envs/gds/lib/python3.5/site-packages/pysal/weights/Distance.py in __init__(self, data, threshold, p, alpha, binary, ids, build_sp, silent)
    788         self._band()
    789         neighbors, weights = self._distance_to_W(ids)
--> 790         W.__init__(self, neighbors, weights, ids, silent_island_warning=self.silent)
    791 
    792     @classmethod

/home/dani/anaconda/envs/gds/lib/python3.5/site-packages/pysal/weights/weights.py in __init__(self, neighbors, weights, id_order, silent_island_warning, ids)
    181         self._reset()
    182         self._n = len(self.weights)
--> 183         if self.islands and not self.silent_island_warning:
    184             ni = len(self.islands)
    185             if ni == 1:

/home/dani/anaconda/envs/gds/lib/python3.5/site-packages/pysal/weights/weights.py in islands(self)
    502         if 'islands' not in self._cache:
    503             self._islands = [i for i,
--> 504                              c in list(self.cardinalities.items()) if c == 0]
    505             self._cache['islands'] = self._islands
    506         return self._islands

/home/dani/anaconda/envs/gds/lib/python3.5/site-packages/pysal/weights/weights.py in cardinalities(self)
    430             c = {}
    431             for i in self._id_order:
--> 432                 c[i] = len(self.neighbors[i])
    433             self._cardinalities = c
    434             self._cache['cardinalities'] = self._cardinalities

KeyError: 49
```

This is under:

```python
In [49]: ps.version
Out[49]: '1.13.0'
```

**Comment by ljwolf** [comment]
I think this occurs in [`_distance_to_W`](https://github.com/pysal/pysal/blob/master/pysal/weights/Distance.py#L877), which ignores the `ids` argument totally.

I have a patch which adds `id_order=ids` to both `WSP` calls and it seems to work. 

I need to unittest this to make sure it's actually working, though.

**Comment by ljwolf** [comment]
fixed by #946, check on your data @darribas
